### PR TITLE
[fix/branding refinement] Branding: Color and UI Improvements

### PR DESCRIPTION
**Pull request by hosy**
## Description
- Setup a complete customer branding with only two color values `darkBrandColor` and `lightBrandColor`
- set all other branding color and style values to optional values
- ownBrander gomplate script was moved into this repository
- show always detailed single account view, instead of server list (if only one account is configured) (also for unbranded app)
- show `Add Account` in detailed single account view, if `can-add-account` is `true`
- added some default colors for branded builds and regrading which `darkBrandColor` or `lightBrandColor` were used
- added a new color style `loginStatusBarStyle` to set a custom status bar style for the login view
- if the key `app.version-number` exists, fastlane will use this value as app and extension short version number, which gives the possibility to set custom version numbers

## Related Issue
https://github.com/owncloud/enterprise/issues/4804
https://github.com/owncloud/enterprise/issues/4759
#1014 
https://github.com/owncloud/enterprise/issues/4828

## Screenshots (if appropriate):

Simulator Screen Shot - iPhone 12 Pro - 2021-10-20 at 15 08 25

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

### CODE REVIEW

File: ownCloudAppShared/AppLock/PasscodeViewController.swift [hunk 1]
```diff
@@ -285,6 +285,14 @@ public class PasscodeViewController: UIViewController, Themeable {
 
 	// MARK: - Themeing
 	public override var preferredStatusBarStyle : UIStatusBarStyle {
+		if VendorServices.shared.isBranded {
+			if #available(iOSApplicationExtension 13.0, *) {
+				return .darkContent
+			} else {
+				return .default
+			}
```
Comment (felix-schwarz) [inline ownCloudAppShared/AppLock/PasscodeViewController.swift:293]: Are these styles really applicable to all brandings? As-is, overriding these isn't possible.

**Review by felix-schwarz — CHANGES_REQUESTED** [review]

File: ownCloudAppShared/User Interface/Theme/UI/ThemeNavigationController.swift [hunk 2]
```diff
@@ -22,6 +22,16 @@ open class ThemeNavigationController: UINavigationController {
 	private var themeToken : ThemeApplierToken?
 
 	override open var preferredStatusBarStyle : UIStatusBarStyle {
+		if let object = self.viewControllers.last {
+			let loginController = ["StaticLoginViewController", "StaticLoginSingleAccountServerListViewController"]
+
+			if self.presentedViewController == nil, loginController.contains(String(describing: type(of: object))) {
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloudAppShared/User Interface/Theme/UI/ThemeNavigationController.swift:28]: Thanks for that hint! I added a protocol to solve the dependency problem and class name problems.

File: ownCloudAppShared/UIKit Extension/UITableViewController+Extension.swift [hunk 3]
```diff
@@ -38,4 +39,52 @@ public extension UITableViewController {
...
+
+	func colorSection(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath, borderColor: UIColor?) {
+		let cornerRadius: CGFloat = 10.0
+		let layer: CAShapeLayer = CAShapeLayer()
+		let pathRef: CGMutablePath = CGMutablePath()
+
+		let bounds: CGRect = cell.bounds.insetBy(dx: 0, dy: 0)
+
+		if indexPath.row == 0 && indexPath.row == tableView.numberOfRows(inSection: indexPath.section) - 1 {
+			pathRef.__addRoundedRect(transform: nil, rect: bounds, cornerWidth: cornerRadius, cornerHeight: cornerRadius)
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloudAppShared/UIKit Extension/UITableViewController+Extension.swift:51]: Awesome, fixed!

File: ownCloudAppShared/User Interface/Theme/ThemeCollection.swift [hunk 4]
```diff
@@ -243,10 +244,15 @@ public class ThemeCollection : NSObject {
 		let rowColor : UIColor? = UIColor.black.withAlphaComponent(0.1)
 		self.tableRowBorderColor = colors.resolveColor("Table.tableRowBorderColor", rowColor)
 
+		var defaultTableRowLabelColor = darkColor
+		if VendorServices.shared.isBranded {
+			defaultTableRowLabelColor = UIColor(hex: 0x000000)
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloudAppShared/User Interface/Theme/ThemeCollection.swift:249]: Yes, this should be default for all brandings, but can be overridden by `Branding.plist`

File: ownCloudAppShared/AppLock/PasscodeViewController.swift (hunk 1 above)
Comment (hosy) [reply ownCloudAppShared/AppLock/PasscodeViewController.swift:293]: I figured out, that the general status bar style does not work for branded apps and can be static for this view. At the moment it cannot be overridden, if this is needed, we can add another parameter to the `ThemeCollection` class.

**Comment by hosy** [comment]
@felix-schwarz all CR findings are fixed or I added a comment and you can give it a new try. Thanks!

**Comment by hosy** [comment]
@jesmrec not sure, if this one need deeper testing by your side? Most things were tested by @NannaBarz and by you in the 11.7.2 preview version for the customers.

**Comment by jesmrec** [comment]
This was already tested. Testing of such colours and branding options has to be done with a global perspective, taking all of them as a whole. This is a task to be scheduled and done somewhen. This specific PR is OK to be moved forward.

**Comment by hosy** [comment]
@jesmrec ok, would be cool if you can approve the PR.

**Review by jesmrec — APPROVED** [review]

**Pull request merged**